stored_requests/backends/file_fetcher/fetcher.go

```diff
@@ -16,45 +16,93 @@ import (
 // This expects each file in the directory to be named "{config_id}.json".
 // For example, when asked to fetch the request with ID == "23", it will return the data from "directory/23.json".
 func NewFileFetcher(directory string) (stored_requests.Fetcher, error) {
-	storedReqData, err := collectStoredData(directory + "/stored_requests")
-	if err != nil {
-		return nil, err
-	}
-	storedImpData, err := collectStoredData(directory + "/stored_imps")
-	if err != nil {
-		return nil, err
-	}
-
-	return &eagerFetcher{storedReqData, storedImpData}, nil
+	storedData, err := collectStoredData(directory, FileSystem{make(map[string]FileSystem), make(map[string]json.RawMessage)}, nil)
```

Comment:
You should make a variant of `collectStoredData()` that will unmarshal the json on startup, so we don't unmarshal on every request. It may even reduce memory requirements to store as a map rather than a raw JSON blob.

Reply:
Another option is to use `collectStoredData()`, but then go through the unmarshalling process on the result to get the maps to finally store in the fetcher.

Reply:
@hhhjort this code is now refactored: 
- Categories unmarshalling is now lazy and works ones per file at the first request to this file
- Unmarshalled categories are now stored in map
- Unmarshalled files are deleted from fileSystem variable after successful unmarshalling